modules/service/src/main/scala/lucuma/odb/service/CalibrationsService.scala

```diff
@@ -74,17 +94,134 @@ object CalibrationsService {
           }
         } else none.pure[F]
 
+      private def calibrationObservations(pid: Program.Id, gid: Group.Id, gnls: List[GmosNConfigs], gsls: List[GmosSConfigs])(using Transaction[F]): F[Result[List[Observation.Id]]] = {
+        def gmosNorthLSObservations(gnls: List[GmosNConfigs]) = gnls.traverse { case (g, f, w, xb, yb) =>
+          val conf =
+            GmosLongSlitInput.Create.North(
+              grating = g,
+              filter = none,
+              fpu = f,
+              common = GmosLongSlitInput.Create.Common(
+                centralWavelength = w,
+                explicitXBin = xb,
+                explicitYBin = yb,
+                explicitAmpReadMode = none,
+                explicitAmpGain = none,
+                explicitRoi = GmosRoi.CentralSpectrum.some,
+                explicitλDithers = none,
+                explicitSpatialOffsets = none
+              )
+            )
+
+          observationService.createObservation(
+            CreateObservationInput(
+              programId = pid.some,
+              proposalReference = none,
+              programReference = none,
+              SET = ObservationPropertiesInput.Create.Default.copy(
+                      group = gid.some,
+                      posAngleConstraint = PosAngleConstraintInput(
+                        mode = PosAngleConstraintMode.AverageParallactic.some, none
+                      ).some,
+                      observingMode = ObservingModeInput.Create(conf.some, none).some,
+                      scienceRequirements =
+                        ScienceRequirementsInput(
+                          mode = ScienceMode.Spectroscopy.some,
+                          spectroscopy = SpectroscopyScienceRequirementsInput.Default.copy(
+                            signalToNoise = Nullable.NonNull(SignalToNoise.unsafeFromBigDecimalExact(100.0))
+                        ).some
+                      ).some
+                    ).some
+            )
+          )
+        }
+        def gmosSouthLSObservations(gsls: List[GmosSConfigs]) = gsls.traverse { case (g, f, w, xb, yb) =>
```

Comment:
Seems like most of this is in common with the `North` version.

Reply:
yes it is very similar. I'm not sure if it  can be parametrized as they are disjoint types